FHEM/00_SIGNALduino.pm

```diff
@@ -2291,21 +2308,17 @@ sub SIGNALduino_Parse_MU($$$$@)
 
 				$valid=1; # Set valid to 1 for every loop
 				#Debug $patternLookupHash{substr($rawData,$i,$signal_width)}; ## Get $signal_width number of chars from raw data string
-				if (exists $patternLookupHash{$sig_str}) 
-				{
-					my $bit = $patternLookupHash{$sig_str};
-					
-					push(@bit_msg,$bit) if (looks_like_number($bit)) ; ## Add the bits to our bit array
+				if (exists $patternLookupHash{$sig_str}) {
+					push(@bit_msg,$patternLookupHash{$sig_str})  ## Add the bits to our bit array
```

Comment:
Ich glaube ich wollte mit if (looks_like_number($bit)) abfangen, dass keine F Bezeichner in die @bit_msg kommen. Ein F ist ja auch kein bit, allerdings bin ich selbst nicht sicher, was dann damit passieren soll.

Reply:
Binär gibt es nur 0 oder 1.

Diese Erfindungen mit F sind doch der größte Mist überhaupt und machen es nur unübersichtlich.
Gibt es denn dafür einen Anwendungsfall, dass hier dieses Sonderformat übergeben wird? Was ich nicht so gut finde ist, dass wir in eine Variable bit_msg etwas speichern das keine bits mehr sind.

Ich würde ja immer erwarten nur valide Bits in so einer Variable zu finden und nicht etwas anderes.

Reply:
Ja, das F = float wird verwendet bei
ID 3.1 - IT remote Control PAR 1000, ITS-150'
ID 82 - Fernotron

bei sendmsg kann ein bit auch mehrere Zustände haben
```
		my @bits = split("", $data);
		my %bitconv = (1=>"one", 0=>"zero", 'D'=> "float", 'F'=> "float", 'P'=> "pause");
```